spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/data/vectorized/VectorizedSparkOrcReaders.java

```diff
@@ -209,7 +229,8 @@ private int numNullsHelper() {
     }
 
     protected int getRowIndex(int rowId) {
-      return vector.isRepeating ? 0 : rowId;
+      int row = isSelectedInUse ? selected[rowId] : rowId;
```

Comment:
will there be a possibility of isSelectedInUse=true and selected is empty?

Reply:
No, this is not a possibility. Similar to other vectors the selected vector is initialized at the start to max batch size and all of the indices in this are excepted to be within the batch size.